src/components/auth/signup.js

```diff
@@ -0,0 +1,122 @@
+import React, { Component } from 'react';
+import { Field, reduxForm } from 'redux-form';
+import { Button, Form, FormGroup, FormFeedback, Label, Input } from 'reactstrap';
+import { connect } from 'react-redux';
+import { Link } from 'react-router-dom';
+
+import CarouselComponent from '../universal/carousel';
+import { signUpAction } from '../../actions';
+
+class Signup extends Component {
+
+    // method for rendering form field on the component
+    renderField(field) {
+        //de-structuring from the field object
+        const { meta: { touched, error } } = field;
+        const isValid = touched && error ? false : '';
+        return (
+            <FormGroup>
+                <Label htmlFor={ field.htmlfor }>{ field.label }</Label>
+                <Input 
+                    type={ field.type }
+                    id={ field.id }
+                    { ...field.input }
+                    valid = { isValid }
+                />
+                <FormFeedback>{ touched ? error : '' }</FormFeedback>
+            </FormGroup>
+        );
+    }
+
+    // method callback on form submission
+    onSubmitForm = (values) => {
```

Comment:
Out of curiosity..what parameter does `values` represent in the context of the function?